marklogic-data-hub/src/test/java/com/marklogic/hub/dataservices/ingestion/BulkIngestTest.java

```diff
@@ -32,8 +32,8 @@ public void setupTest() {
     public void testBulkIngest() {
 
         String prefix = "/bulkIngesterTest";
-        String endpointState = "{\"next\":" + 0 + ", \"uriprefix\":\""+prefix+"\"}";
-        String workUnit      = "{\"taskId\":"+1+"}";
+        String endpointState =  "{}";
+        String workUnit      = "{\"userDefinedValue\":" + 1 + ", \"uriprefix\":\""+prefix+"\"}";
```

Comment:
what's the purpose of userDefinedValue?

Reply:
"userDefinedValue" is just a variable added by the user. We can also have something like the below -
String workUnit      = "{\"testValue\":" + 1 + ", \"uriprefix\":\""+prefix+"\"}";

Adding the "userDefinedValue" confirms that no exception is thrown even when we have extra values defined by the user.

Reply:
How will the user pass this option from glue/spark? I'm still confused where are we using this?

Reply:
This test is not in the marklogic-data-hub-spark-connector folder. It is testing only the bulk caller in the java-client-api and uses the bulkIngester api and sjs. The uriPrefix appended in the workUnit is used by the sjs file here - https://github.com/anu3990/marklogic-data-hub/blob/660d87d7616098274a5b97ff3d2005b0a07461fb/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkIngester.sjs#L39 

In the Glue environment, uriprefix will be passed in as an option(i have defined in the next comment) and appended in the workUnit by the HubDataWriter class before calling bulkIngester.api.

Reply:
I understand that. I'm confused on the userDefinedValue. How it is passed and what's the use of it?

Reply:
We don't need the UserDefinedValue here.